### PR TITLE
Fix #2419: Updated navigation drawer tests in HelperFragmentTest

### LINKED ISSUE #2257: Remove unnecessary Initialisation of  Firebase app in Testing
**Describe the bug**
Most of the places in the app module testing files we are initializing Firebase app, which is used previously for some workaround but it is not needed now. 

**Expected behavior**
1. Remove the initialization - `FirebaseApp.initializeApp(context)`
2. Run the test in Espresso and Robolectric and see if all the test cases are passing or not. 
3. If all the test cases are failing without `FirebaseApp.initializeApp(context)`, we don't need to remove. 

**Usage**
Screenshot 2020-12-14 at 14 23 39

**Additional context**
[How to run the test on Robolectric?](https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing#running-robolectric-tests)

### LINKED ISSUE #2419: Fix Navigation Drawer test in HelpFragmentTest [Testing]
**Describe the bug**
`openHelp_openNavDrawerAndClose_closingOfNavDrawerIsVerifiedSuccess` this test is not correct for Robolectric.

**Expected behavior**
Use a similar solution, test on both espresso and robolectric and add screenshot of test result in pr.
https://github.com/oppia/oppia-android/blob/83e409a64bbff8c1e724c4e52dfed71f8a70ec40/app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerActivityTest.kt#L412

Steps:
1. Check the correctness of the above-mentioned test
2. Find and fix all other similar issues 

```kotlin
@Test
  fun openHelp_openNavDrawerAndClose_closingOfNavDrawerIsVerifiedSuccess() {
    launch<HelpActivity>(createHelpActivityIntent(0, true)).use {
      it.openNavigationDrawer()
      onView(withId(R.id.help_activity_drawer_layout)).perform(close())
      onView(withId(R.id.help_activity_drawer_layout)).check(matches(isClosed()))
    }
  }

  private fun ActivityScenario<HelpActivity>.openNavigationDrawer() {
    onView(withContentDescription(R.string.drawer_open_content_description))
      .check(matches(isCompletelyDisplayed()))
      .perform(click())

    // Force the drawer animation to start. See https://github.com/oppia/oppia-android/pull/2204 for
    // background context.
    onActivity { activity ->
      val drawerLayout =
        activity.findViewById<DrawerLayout>(R.id.home_activity_drawer_layout)
      // Note that this only initiates a single computeScroll() in Robolectric. Normally, Android
      // will compute several of these across multiple draw calls, but one seems sufficient for
      // Robolectric. Note that Robolectric is also *supposed* to handle the animation loop one call
      // to this method initiates in the view choreographer class, but it seems to not actually
      // flush the choreographer per observation. In Espresso, this method is automatically called
      // during draw (and a few other situations), but it's fine to call it directly once to kick it
      // off (to avoid disparity between Espresso/Robolectric runs of the tests).
      // NOTE TO DEVELOPERS: if this ever flakes, we can probably put this in a loop with fake time
      // adjustments to simulate the render loop.
      drawerLayout.computeScroll()
    }

    // Wait for the drawer to fully open (mostly for Espresso since Robolectric should synchronously
    // stabilize the drawer layout after the previous logic completes).
    testCoroutineDispatchers.runCurrent()
  }
```

### PR DESCRIPTION
**Pull request by srikanthkb**
## Explanation

Fixes #2419 
Fixes #2257

Updated a new method to be used for navigation drawer opening in the `HelpFragmentTest` and `OptionsFragmentTest`, to support RoboElectric tests.
Screenshot of the tests with the changes is attached below:

image

image

Removed unnecessary `FirebaseApp.initialization()` from the test files to support Roboelectric and Esspresso tests completion.
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/sharedTest/java/org/oppia/android/app/help/HelpFragmentTest.kt [hunk 1]
```diff
@@ -171,12 +171,35 @@ class HelpFragmentTest {
...
+      // will compute several of these across multiple draw calls, but one seems sufficient for
+      // Robolectric. Note that Robolectric is also *supposed* to handle the animation loop one call
+      // to this method initiates in the view choreographer class, but it seems to not actually
+      // flush the choreographer per observation. In Espresso, this method is automatically called
+      // during draw (and a few other situations), but it's fine to call it directly once to kick it
+      // off (to avoid disparity between Espresso/Robolectric runs of the tests).
+      // NOTE TO DEVELOPERS: if this ever flakes, we can probably put this in a loop with fake time
+      // adjustments to simulate the render loop.
+      drawerLayout.computeScroll()
+    }
```
Comment (srikanthkb) [inline app/src/sharedTest/java/org/oppia/android/app/help/HelpFragmentTest.kt:213]: @anandwana001 
I have not included the `testCoroutineDispatchers.runCurrent()` here.
Is it required ? if yes, we have to inject the `testCoroutineDispatchers` into HelperFragmentTest

Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/android/app/help/HelpFragmentTest.kt:213]: Let's add it, this is something which stabilizes the test. 
Could you please also add a screenshot for robolectric test result?

**Comment by anandwana001** [comment]
Let's find if any other test fil has a similar issue.

**Comment by rt4914** [comment]
@Srikanth-Kb Make sure that the `Assignees` section is up-to-date. 
https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section

**Comment by srikanthkb** [comment]
> 
> 
> @Srikanth-Kb Make sure that the `Assignees` section is up-to-date.
> https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section

Noted. Thanks.

**Review by rt4914 — APPROVED** [review]
LGTM

**Review by anandwana001 — CHANGES_REQUESTED** [review]
`testOptionFragment_clickNavigationDrawerHamburger_navigationDrawerIsOpenedSuccessfully`

need to fix this test case as well.

**Comment by srikanthkb** [comment]
> `testOptionFragment_clickNavigationDrawerHamburger_navigationDrawerIsOpenedSuccessfully`
> 
> need to fix this test case as well.

Got it.. I will check for other occurrences and update. Thank you @anandwana001

**Comment by anandwana001** [comment]
@Srikanth-Kb any update on this?

**Comment by srikanthkb** [comment]
> @Srikanth-Kb any update on this?

Updated the PR with changes.

**Comment by srikanthkb** [comment]
@anandwana001 requesting your review.

**Review by anandwana001 — APPROVED** [review]
LGTM

**Pull request merged**